sleap_nn/data/sleap_dataset.py

```diff
@@ -0,0 +1,70 @@
+"""Handle distribution of sleap data."""
+from typing import Sequence, Tuple, Text, Union, Optional, List
+
+import torch
+import numpy as np
+from torch import nn
+from torch.utils.data import Dataset, DataLoader
+import lightning.pytorch as pl
+
+import sleap_io as sio
+
+
+class SleapDataset(Dataset):
+    """Dataset for loading items from Labels object."""
+
+    def __init__(
+        self,
+        labels: sio.Labels,
+        output_stride: int = 2,
+        sigma: int = 2.5,
+    ):
+        """Initialize SleapDataset.
+
+        Args:
+            labels: sleap_io.Labels object that contains LabeledFrames
+            output_stride: Relative stride of the generated confidence maps. This is
+                effectively the reciprocal of the output scale, i.e., increase this to
+                generate confidence maps that are smaller than the input images.
+            sigma: Standard deviation of the 2D Gaussian distribution sampled to
+                generate confidence maps. This defines the spread in units of the input
+                image’s grid, i.e., it does not take scaling in previous steps into account.
+        """
+        self.labels = labels
+        self.sigma = sigma
+        self.output_stride = output_stride
+        self.data = []
+
+        """
+        Store data in self.data -> format [(LabeledFrame index, Instance index), 
+            (LabeledFrame index, Next Instance index)...]
+        """
```

Comment:
`"""` comment blocks should not be inline (reserved for docstrings). Use `#` for inline comments, even if they're multi-line